docs/developers/tooling/cross-chain/ccip-read-gateway.mdx

```diff
@@ -0,0 +1,120 @@
+---
+title: CCIP Read
+description: Use a CCIP Read gateway to query data on Linea from L1
+---
+
+## What is CCIP Read?
+
+CCIP Read, closely associated with [EIP-3668](https://eips.ethereum.org/EIPS/eip-3668) and also 
+referred to by its full name [Cross-chain Interoperability Protocol](https://chain.link/cross-chain), 
+is a protocol developed by Chainlink to enable applications to access offchain data (i.e. on an L2) 
+through a kind of server known as a gateway. 
+
+CCIP Read has been implemented by ENS to enable domains to resolve on L1 even if the relevant data 
+and records are stored on an L2 chain — an offchain resolver framework. The Linea team adapted the 
+functionality of the relevant ENS contract, [`evm-gateway`](https://github.com/ensdomains/evmgateway),
+so that it would function correctly with Linea's Sparse Merkle Tree design. The Linea ENS repository
+therefore adds support for CCIP Read to Linea.
+
+:::info 
+
+Read more about Sparse Merkle Trees in our [architecture documentation](/architecture/stack/evm-state-manager).
+
+:::
+
+### Linea ENS
+
+CCIP Read was introduced as a necessary part of Linea ENS — the very first full implementation of 
+Ethereum Name Service (ENS) on an L2. 
```

Comment:
Agree, also depending how someone interprets "first full implementation" it might also not completely be accurate. We are first zkEVM Layer 2 to adopt ENS EIP-3668 standard in production